presto-system-access-control/src/main/java/com/facebook/presto/security/KerberosExactMatchAccessControl.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.security;
+
+import com.facebook.presto.spi.CatalogSchemaName;
+import com.facebook.presto.spi.CatalogSchemaTableName;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.security.Identity;
+import com.facebook.presto.spi.security.Privilege;
+import com.facebook.presto.spi.security.SystemAccessControl;
+import com.google.common.base.Splitter;
+
+import java.security.Principal;
+import java.util.List;
+
+import static com.facebook.presto.spi.StandardErrorCode.PERMISSION_DENIED;
+
+public class KerberosExactMatchAccessControl
+        implements SystemAccessControl
+{
+    @Override
+    public void checkCanSetUser(Principal principal, String userName)
+    {
+        List<String> parts = Splitter.on("@").splitToList(principal.getName());
```

Comment:
This also needs to correctly handle `\` escaped `@` symbols in the principal name. I'll test that too.